### PR TITLE
Latest winget

### PR DESCRIPTION
**Pull request by KnifMelti**
# Proposed Changes
Now with the constant developing/releases of WinGet we need to update all the time...

## Related Issues
https://github.com/Romanitho/Winget-Install/discussions/89

### CODE REVIEW

**Comment by Romanitho** [comment]
Maybe we can use your Get-AvailableWinGetVersion function during install and post install tasks?

**Comment by KnifMelti** [comment]
> Maybe we can use your Get-AvailableWinGetVersion function during install and post install tasks?

Does that in Post alteady
`$AvailableWinGetVersion = Get-AvailableWinGetVersion`

File: Winget-AutoUpdate-Install.ps1 [hunk 1]
```diff
@@ -118,6 +118,22 @@ param(
...
+    try {
+        #Return latest version
+        return ((Invoke-WebRequest $WinGeturl -UseBasicParsing | ConvertFrom-Json)[0].tag_name).Replace("v", "")
+    }
+    catch {
+        return $false
+    }
+
+}
+
```
Comment (Romanitho) [inline Winget-AutoUpdate-Install.ps1:136]: I mean here

Comment (KnifMelti) [reply Winget-AutoUpdate-Install.ps1:136]: Yes, but the import of functions doesn't exist in `Winget-AutoUpdate-Install.ps1`, hence the Function inline...
...or do I not understand?

Comment (KnifMelti) [reply Winget-AutoUpdate-Install.ps1:136]: Ok, forgot about that. I'll fix it when  I get home

Comment (Romanitho) [reply Winget-AutoUpdate-Install.ps1:136]: it's better, I think. Thanks

**Comment by Romanitho** [comment]
👌🏼

File: Winget-AutoUpdate-Install.ps1 [hunk 2]
```diff
@@ -181,20 +181,31 @@ function Install-Prerequisites {
...
-        Write-Host "Winget is Installed" -ForegroundColor Green
+    #Check available WinGet version, if fail set version to the latest version as of 2023-10-08
+    . "$PSScriptRoot\Winget-AutoUpdate\functions\Get-AvailableWinGetVersion.ps1"
+    $AvailableWinGetVersion = Get-AvailableWinGetVersion
+    if (!$AvailableWinGetVersion) {
+        $AvailableWinGetVersion = "1.6.2771"
+    }
 
+    #Check if WinGet is installed, if not set version to dummy...
+    $ResolveWingetPath = Resolve-Path "$env:programfiles\WindowsApps\Microsoft.DesktopAppInstaller_*_*__8wekyb3d8bbwe\winget.exe" | Sort-Object { [version]($_.Path -replace '^[^\d]+_((\d+\.)*\d+)_.*', '$1') }
```
[the remark this answers is not in the document]
Comment (KnifMelti) [reply Winget-AutoUpdate-Install.ps1:194]: I don't understand, but do as is best!

[the remark this answers is not in the document]
Comment (Romanitho) [reply Winget-AutoUpdate-Install.ps1:194]: 😅

**Pull request merged**